### PR TITLE
feat: support selection of sector (issue-8)

### PR DESCRIPTION
**Pull request by brennan3**

### CODE REVIEW

File: bots/reads.ts [hunk 1]
```diff
@@ -16,6 +16,12 @@ export interface ReadsConfig {
   botToken: string;
   secretToken: string;
   webhookUrl: string;
+  sectors: Sector[]
```
Comment (stocks29) [inline bots/reads.ts]:
```suggestion
  sectors: Sector[];
```

File: bots/reads.ts [hunk 2]
```diff
@@ -155,6 +165,17 @@ const displayMenu = async (ctx: ReadsContext) => {
   await ctx.reply('What would you like to do?', buttons);
 };
 
+const displaySectorMenu = async (ctx: ReadsContext, sectors: Sector[]) => {
+  const buttonRows = [];
+  for (let i = 0; i < sectors.length; i+= 2) {
+    const chunk = sectors.slice(i, i + 2);
+    const sectorRowButtons = chunk.map(({ slug, title }) => {return Markup.button.callback(title, `setsector_${slug}`)});
```
Comment (stocks29) [inline bots/reads.ts]:
```suggestion
    const sectorRowButtons = chunk.map(({ slug, title }) => Markup.button.callback(title, `setsector_${slug}`));
```

File: server.ts [hunk 3]
```diff
@@ -18,6 +18,14 @@ if (!DELPHI_READS_WEBHOOK_URL) throw new Error('"DELPHI_READS_WEBHOOK_URL" env v
 const app = express();
 const port = DELPHI_READS_BOT_PORT || 6000;
 
+const sectors: Sector[] = [
+  { slug: 'All Sectors', title: 'All Sectors' },
```
Comment (stocks29) [inline server.ts]: on the FE and TG summary we called this `general`. might be nice to be consistent.

also, I think you need to map this to all the real sectors when populating `item.taxonomy`.

File: bots/reads.ts [hunk 4]
```diff
@@ -272,30 +293,36 @@ const handleUpdateUrl = async (url: string, ctx: ReadsContext, config: ReadsConf
...
   bot.action('post', handlePost);
   bot.action('setdescription', handleSetDescription);
-  bot.action('setsector', handleSetTaxonomy);
   bot.action('settitle', handleSetTitle);
   bot.action('settype', handleSetTag);
+  bot.action('setsector', async (ctx) => { await handleSetTaxonomy(ctx, config.sectors)});
+
+  // dynamic actions
+  bot.action(/setsector_(.+)/, async (ctx) => {
+    ctx.session.item.taxonomy = [getSectorLabel(config.sectors, ctx.match[1])];
```
Comment (stocks29) [inline bots/reads.ts:323]: need to translate `all` to all the real sectors

Comment (brennan3) [reply bots/reads.ts:323]: Ended up just naming it `general` and will transpose on the actual API call

**Pull request merged**